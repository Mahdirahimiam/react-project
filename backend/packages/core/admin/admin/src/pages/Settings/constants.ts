import { MenuItem } from '@strapi/helper-plugin';

export interface Route
  extends Pick<MenuItem, 'exact' | 'to'>,
    Required<Pick<MenuItem, 'Component'>> {}

export const ROUTES_CE: Route[] = [
  {
    async Component() {
      const { ProtectedListPage } = await import('./pages/Roles/ListPage');

      return ProtectedListPage;
    },
    to: '/settings/roles',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedCreatePage } = await import('./pages/Roles/CreatePage');

      return ProtectedCreatePage;
    },
    to: '/settings/roles/duplicate/:id',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedCreatePage } = await import('./pages/Roles/CreatePage');

      return ProtectedCreatePage;
    },
    to: '/settings/roles/new',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedEditPage } = await import('./pages/Roles/EditPage');

      return ProtectedEditPage;
    },
    to: '/settings/roles/:id',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedListPage } = await import('./pages/Users/ListPage');

      return ProtectedListPage;
    },
    to: '/settings/users',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedEditPage } = await import('./pages/Users/EditPage');

      return ProtectedEditPage;
    },
    to: '/settings/users/:id',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedCreatePage } = await import('./pages/Webhooks/CreatePage');

      return ProtectedCreatePage;
    },
    to: '/settings/webhooks/create',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedEditPage } = await import('./pages/Webhooks/EditPage');

      return ProtectedEditPage;
    },
    to: '/settings/webhooks/:id',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedListPage } = await import('./pages/Webhooks/ListPage');

      return ProtectedListPage;
    },
    to: '/settings/webhooks',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedListView } = await import('./pages/ApiTokens/ListView');

      return ProtectedListView;
    },
    to: '/settings/api-tokens',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedCreateView } = await import('./pages/ApiTokens/CreateView');

      return ProtectedCreateView;
    },
    to: '/settings/api-tokens/create',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedEditView } = await import('./pages/ApiTokens/EditView/EditViewPage');

      return ProtectedEditView;
    },
    to: '/settings/api-tokens/:id',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedCreateView } = await import('./pages/TransferTokens/CreateView');

      return ProtectedCreateView;
    },
    to: '/settings/transfer-tokens/create',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedListView } = await import('./pages/TransferTokens/ListView');

      return ProtectedListView;
    },
    to: '/settings/transfer-tokens',
    exact: true,
  },
  {
    async Component() {
      const { ProtectedEditView } = await import('./pages/TransferTokens/EditView');

      return ProtectedEditView;
    },
    to: '/settings/transfer-tokens/:id',
    exact: true,
  },
  {
    async Component() {
      const { PurchaseAuditLogs } = await import('./pages/PurchaseAuditLogs');

      return PurchaseAuditLogs;
    },
    to: '/settings/purchase-audit-logs',
    exact: true,
  },
  {
    async Component() {
      const { PurchaseReviewWorkflows } = await import('./pages/PurchaseReviewWorkflows');

      return PurchaseReviewWorkflows;
    },
    to: '/settings/purchase-review-workflows',
    exact: true,
  },
  {
    async Component() {
      const { PurchaseSingleSignOn } = await import('./pages/PurchaseSingleSignOn');

      return PurchaseSingleSignOn;
    },
    to: '/settings/purchase-single-sign-on',
    exact: true,
  },
];
