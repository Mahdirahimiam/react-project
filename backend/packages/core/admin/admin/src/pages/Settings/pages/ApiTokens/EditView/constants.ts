import { translatedErrors } from '@strapi/helper-plugin';
import * as yup from 'yup';

export const schema = yup.object().shape({
  name: yup.string().max(100).required(translatedErrors.required),
  type: yup
    .string()
    .oneOf(['read-only', 'full-access', 'custom'])
    .required(translatedErrors.required),
  description: yup.string().nullable(),
  lifespan: yup.number().integer().min(0).nullable().defined(translatedErrors.required),
});
