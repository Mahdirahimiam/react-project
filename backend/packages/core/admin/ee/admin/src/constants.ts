import { AuthResponse } from './pages/AuthResponse';

import type { SettingsMenu } from '../../../admin/src/constants';
import type { PermissionMap } from '../../../admin/src/types/permissions';

export const ADMIN_PERMISSIONS_EE = {
  settings: {
    auditLogs: {
      main: [{ action: 'admin::audit-logs.read', subject: null }],
      read: [{ action: 'admin::audit-logs.read', subject: null }],
      update: [{ action: 'admin::audit-logs.update', subject: null }],
    },
    'review-workflows': {
      main: [{ action: 'admin::review-workflows.read', subject: null }],
      read: [{ action: 'admin::review-workflows.read', subject: null }],
      create: [{ action: 'admin::review-workflows.create', subject: null }],
      delete: [{ action: 'admin::review-workflows.delete', subject: null }],
      update: [{ action: 'admin::review-workflows.update', subject: null }],
    },
    sso: {
      main: [{ action: 'admin::provider-login.read', subject: null }],
      read: [{ action: 'admin::provider-login.read', subject: null }],
      update: [{ action: 'admin::provider-login.update', subject: null }],
    },
  },
} satisfies {
  settings: Pick<PermissionMap['settings'], 'auditLogs' | 'review-workflows' | 'sso'>;
};

export const ROUTES_EE = [
  {
    Component: () => ({ default: AuthResponse }),
    to: '/auth/login/:authResponse',
    exact: true,
  },
];

// TODO: the constants.js file is imported before the React application is setup and
// therefore `window.strapi` might not exist at import-time. We should probably define
// which constant is available at which stage of the application lifecycle.
export const SETTINGS_LINKS_EE = (): SettingsMenu => ({
  global: [
    ...(window.strapi.features.isEnabled(window.strapi.features.SSO)
      ? [
          {
            intlLabel: { id: 'Settings.sso.title', defaultMessage: 'Single Sign-On' },
            to: '/settings/single-sign-on',
            id: 'sso',
          },
        ]
      : []),

    ...(window.strapi.features.isEnabled(window.strapi.features.REVIEW_WORKFLOWS)
      ? [
          {
            intlLabel: {
              id: 'Settings.review-workflows.page.title',
              defaultMessage: 'Review Workflows',
            },
            to: '/settings/review-workflows',
            id: 'review-workflows',
          },
        ]
      : []),
  ],

  admin: [
    ...(window.strapi.features.isEnabled(window.strapi.features.AUDIT_LOGS)
      ? [
          {
            intlLabel: { id: 'global.auditLogs', defaultMessage: 'Audit Logs' },
            to: '/settings/audit-logs?pageSize=50&page=1&sort=date:DESC',
            id: 'auditLogs',
          },
        ]
      : []),
  ],
});
