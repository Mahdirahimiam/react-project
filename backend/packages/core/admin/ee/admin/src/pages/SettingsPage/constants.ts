import type { Route } from '../../../../../admin/src/pages/Settings/constants';

export const ROUTES_EE: Route[] = [
  ...(window.strapi.features.isEnabled(window.strapi.features.AUDIT_LOGS)
    ? [
        {
          async Component() {
            const { ProtectedListPage } = await import('./pages/AuditLogs/ListPage');

            return ProtectedListPage;
          },
          to: '/settings/audit-logs',
          exact: true,
        },
      ]
    : []),

  ...(window.strapi.features.isEnabled(window.strapi.features.REVIEW_WORKFLOWS)
    ? [
        {
          async Component() {
            const { ProtectedReviewWorkflowsPage } = await import(
              './pages/ReviewWorkflows/ListPage'
            );

            return ProtectedReviewWorkflowsPage;
          },
          to: '/settings/review-workflows',
          exact: true,
        },

        {
          async Component() {
            const { ReviewWorkflowsCreatePage } = await import(
              './pages/ReviewWorkflows/CreatePage'
            );

            return ReviewWorkflowsCreatePage;
          },
          to: '/settings/review-workflows/create',
          exact: true,
        },

        {
          async Component() {
            const { ReviewWorkflowsEditPage } = await import('./pages/ReviewWorkflows/EditPage');

            return ReviewWorkflowsEditPage;
          },
          to: '/settings/review-workflows/:workflowId',
          exact: true,
        },
      ]
    : []),

  ...(window.strapi.features.isEnabled(window.strapi.features.SSO)
    ? [
        {
          async Component() {
            const { ProtectedSSO } = await import('./pages/SingleSignOnPage');

            return ProtectedSSO;
          },
          to: '/settings/single-sign-on',
          exact: true,
        },
      ]
    : []),
];
