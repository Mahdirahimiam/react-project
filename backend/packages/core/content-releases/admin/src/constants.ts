import { Permission as StrapiPermission } from '@strapi/helper-plugin';

export const PERMISSIONS = {
  main: [
    {
      action: 'plugin::content-releases.read',
      subject: null,
      id: '',
      actionParameters: {},
      properties: {},
      conditions: [],
    },
  ],
  create: [
    {
      action: 'plugin::content-releases.create',
      subject: null,
      id: '',
      actionParameters: {},
      properties: {},
      conditions: [],
    },
  ],
  update: [
    {
      action: 'plugin::content-releases.update',
      subject: null,
      id: '',
      actionParameters: {},
      properties: {},
      conditions: [],
    },
  ],
  delete: [
    {
      action: 'plugin::content-releases.delete',
      subject: null,
      id: '',
      actionParameters: {},
      properties: {},
      conditions: [],
    },
  ],
  createAction: [
    {
      action: 'plugin::content-releases.create-action',
      subject: null,
      id: '',
      actionParameters: {},
      properties: {},
      conditions: [],
    },
  ],
  deleteAction: [
    {
      action: 'plugin::content-releases.delete-action',
      subject: null,
      id: '',
      actionParameters: {},
      properties: {},
      conditions: [],
    },
  ],
  publish: [
    {
      action: 'plugin::content-releases.publish',
      subject: null,
      id: '',
      actionParameters: {},
      properties: {},
      conditions: [],
    },
  ],
} satisfies Record<string, StrapiPermission[]>;
