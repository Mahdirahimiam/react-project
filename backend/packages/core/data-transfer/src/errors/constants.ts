import { ErrorDiagnosticSeverity } from '../engine/diagnostic';

export const SeverityKind: Record<string, ErrorDiagnosticSeverity> = {
  FATAL: 'fatal',
  ERROR: 'error',
  SILLY: 'silly',
} as const;
export type Severity = (typeof SeverityKind)[keyof typeof SeverityKind];
