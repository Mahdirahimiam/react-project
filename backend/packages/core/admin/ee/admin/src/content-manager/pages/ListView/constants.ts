import { getTranslation } from '../../../../../../admin/src/content-manager/utils/translations';
import { ASSIGNEE_ATTRIBUTE_NAME, STAGE_ATTRIBUTE_NAME } from '../EditView/components/constants';

import { AssigneeFilter } from './components/AssigneeFilter';
import { StageFilter } from './components/StageFilter';

import type { TableHeader } from '../../../../../../admin/src/content-manager/pages/ListView/ListViewPage';
import type { FilterData } from '@strapi/helper-plugin';
import type { MessageDescriptor } from 'react-intl';

interface NonTranslatedTableHeader extends Omit<TableHeader, 'metadatas'> {
  metadatas: Omit<TableHeader['metadatas'], 'label'> & {
    label: MessageDescriptor;
  };
}

export const REVIEW_WORKFLOW_COLUMNS_EE = [
  {
    key: `__${STAGE_ATTRIBUTE_NAME}_temp_key__`,
    name: STAGE_ATTRIBUTE_NAME,
    fieldSchema: {
      type: 'relation',
      relation: 'oneToMany',
      target: 'admin::review-workflow-stage',
    },
    metadatas: {
      // formatMessage() will be applied when the column is rendered
      label: {
        id: getTranslation(`containers.ListPage.table-headers.reviewWorkflows.stage`),
        defaultMessage: 'Review stage',
      },
      searchable: false,
      sortable: true,
      mainField: {
        name: 'name',
        type: 'string',
      },
    },
  },
  {
    key: `__${ASSIGNEE_ATTRIBUTE_NAME}_temp_key__`,
    name: ASSIGNEE_ATTRIBUTE_NAME,
    fieldSchema: {
      type: 'relation',
      target: 'admin::user',
      relation: 'oneToMany',
    },
    metadatas: {
      label: {
        id: getTranslation(`containers.ListPage.table-headers.reviewWorkflows.assignee`),
        defaultMessage: 'Assignee',
      },
      searchable: false,
      sortable: true,
      mainField: {
        name: 'firstname',
        type: 'string',
      },
    },
  },
] satisfies NonTranslatedTableHeader[];

export const REVIEW_WORKFLOW_FILTERS = [
  {
    fieldSchema: {
      type: 'relation',
      mainField: {
        name: 'name',
        type: 'string',
      },
    },

    metadatas: {
      customInput: StageFilter,

      label: {
        id: getTranslation(`containers.ListPage.table-headers.reviewWorkflows.stage`),
        defaultMessage: 'Review stage',
      },
    },

    name: 'strapi_stage',
  },

  {
    fieldSchema: {
      type: 'relation',
      mainField: {
        name: 'id',
        type: 'integer',
      },
    },

    metadatas: {
      customInput: AssigneeFilter,

      customOperators: [
        {
          intlLabel: {
            id: 'components.FilterOptions.FILTER_TYPES.$eq',
            defaultMessage: 'is',
          },
          value: '$eq',
        },
        {
          intlLabel: {
            id: 'components.FilterOptions.FILTER_TYPES.$ne',
            defaultMessage: 'is not',
          },
          value: '$ne',
        },
      ],

      label: {
        id: getTranslation(`containers.ListPage.table-headers.reviewWorkflows.assignee.label`),
        defaultMessage: 'Assignee',
      },
    },

    name: 'strapi_assignee',
  },
] satisfies Array<
  Omit<FilterData, 'metadatas'> & {
    metadatas: Omit<FilterData['metadatas'], 'label'> & {
      label: MessageDescriptor;
    };
  }
>;
