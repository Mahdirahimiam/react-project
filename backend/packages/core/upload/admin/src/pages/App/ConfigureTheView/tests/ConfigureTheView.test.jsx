import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { TrackingProvider } from '@strapi/helper-plugin';
import { fireEvent, render as renderRTL, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';

import ConfigureTheView from '..';
import { pageSizes, sortOptions } from '../../../../constants';

const mutateAsync = jest.fn();
jest.mock('../../../../hooks/useConfig', () => ({
  useConfig: jest.fn(() => ({
    mutateConfig: {
      mutateAsync,
    },
  })),
}));
jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useNotification: jest.fn(() => jest.fn()),
}));

const render = (
  config = {
    pageSize: pageSizes[0],
    sort: sortOptions[0].value,
  }
) => ({
  user: userEvent.setup(),
  ...renderRTL(<ConfigureTheView config={config} />, {
    wrapper: ({ children }) => (
      <IntlProvider locale="en" messages={{}}>
        <TrackingProvider>
          <ThemeProvider theme={lightTheme}>
            <MemoryRouter>{children}</MemoryRouter>
          </ThemeProvider>
        </TrackingProvider>
      </IntlProvider>
    ),
  }),
});

describe('Upload - Configure', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('initial render', () => {
    it('renders and matches the snapshot', () => {
      const { container, getByRole, getByText } = render();

      expect(getByRole('main')).toHaveFocus();
      expect(getByText('Configure the view - Media Library')).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });
  });

  describe('save', () => {
    it('save renders and is initially disabled', () => {
      const { getByText, getByRole } = render();

      expect(getByText('Configure the view - Media Library')).toBeInTheDocument();
      expect(
        getByRole('button', {
          name: 'Save',
        })
      ).toHaveAttribute('disabled');
    });
  });

  describe('user actions', () => {
    const testPageSize = pageSizes[1];

    it('modify settings', async () => {
      const { user, getByRole, getByText } = render();

      expect(getByRole('combobox', { name: 'Entries per page' })).toHaveTextContent('10');

      await user.click(getByRole('combobox', { name: 'Entries per page' }));
      await user.click(getByRole('option', { name: testPageSize.toString() }));

      expect(getByRole('combobox', { name: 'Entries per page' })).toHaveTextContent('20');

      expect(
        getByRole('button', {
          name: 'Save',
        })
      ).toBeEnabled();

      /**
       * using `userEvent.click` does not fire the submit event for the form :(
       * see – https://github.com/testing-library/user-event/issues/1075
       * see – https://github.com/testing-library/user-event/issues/1002
       */
      fireEvent.click(
        getByRole('button', {
          name: 'Save',
        })
      );

      await waitFor(() => {
        expect(getByText('This will modify all your settings')).toBeInTheDocument();
      });

      await user.click(getByText('Confirm'));

      expect(mutateAsync).toHaveBeenCalledTimes(1);
      expect(mutateAsync).toHaveBeenCalledWith({
        pageSize: testPageSize,
        sort: 'createdAt:DESC',
      });
    });
  });
});
