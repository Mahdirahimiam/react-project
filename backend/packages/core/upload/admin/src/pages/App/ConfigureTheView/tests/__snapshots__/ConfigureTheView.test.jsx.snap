// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload - Configure initial render renders and matches the snapshot 1`] = `
.c36 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c8 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #4945ff;
}

.c12 {
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
  color: #32324d;
}

.c17 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c18 {
  font-size: 1rem;
  line-height: 1.5;
  color: #666687;
}

.c24 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c31 {
  font-size: 0.875rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c35 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c1 {
  padding-bottom: 56px;
}

.c4 {
  background: #f6f6f9;
  padding-top: 24px;
  padding-right: 56px;
  padding-bottom: 40px;
  padding-left: 56px;
}

.c5 {
  padding-bottom: 8px;
}

.c10 {
  min-width: 0;
}

.c13 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c19 {
  padding-right: 56px;
  padding-left: 56px;
}

.c20 {
  background: #ffffff;
  padding-top: 24px;
  padding-right: 32px;
  padding-bottom: 24px;
  padding-left: 32px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c26 {
  background: #ffffff;
  padding-top: 4px;
  padding-right: 12px;
  padding-bottom: 4px;
  padding-left: 12px;
  border-radius: 4px;
  position: relative;
  overflow: hidden;
  width: 100%;
  cursor: default;
}

.c29 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c23 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c27 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c30 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 12px;
}

.c15 {
  position: relative;
  outline: none;
}

.c15 > svg {
  height: 12px;
  width: 12px;
}

.c15 > svg > g,
.c15 > svg path {
  fill: #ffffff;
}

.c15[aria-disabled='true'] {
  pointer-events: none;
}

.c15:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c15:focus-visible {
  outline: none;
}

.c15:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c16 {
  height: 2rem;
}

.c16 svg {
  height: 0.75rem;
  width: auto;
}

.c16[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true'] .c7 {
  color: #666687;
}

.c16[aria-disabled='true'] svg > g,
.c16[aria-disabled='true'] svg path {
  fill: #666687;
}

.c16[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true']:active .c7 {
  color: #666687;
}

.c16[aria-disabled='true']:active svg > g,
.c16[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c16:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c16:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c16 svg > g,
.c16 svg path {
  fill: #ffffff;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c28 {
  border: 1px solid #dcdce4;
  min-height: 2.5rem;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c28[aria-disabled='true'] {
  color: #666687;
}

.c28:focus-visible {
  outline: none;
}

.c28:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c34 > svg {
  width: 0.375rem;
}

.c34 > svg > path {
  fill: #666687;
}

.c32 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c37[data-state='checked'] .c7 {
  font-weight: bold;
  color: #4945ff;
}

.c21 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 16px;
}

.c22 {
  grid-column: span 6;
  max-width: 100%;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr;
}

.c2 {
  overflow-x: hidden;
}

.c3:focus-visible {
  outline: none;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  gap: 8px;
  position: relative;
  outline: none;
}

.c6 svg {
  font-size: 0.625rem;
}

.c6 svg path {
  fill: #4945ff;
}

.c6:hover {
  color: #7b79ff;
}

.c6:active {
  color: #271fe0;
}

.c6:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c6:focus-visible {
  outline: none;
}

.c6:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

@media (max-width:68.75rem) {
  .c22 {
    grid-column: span 12;
  }
}

@media (max-width:34.375rem) {
  .c22 {
    grid-column: span;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <main
        aria-labelledby="main-content-title"
        class="c3"
        id="main-content"
        tabindex="-1"
      >
        <form>
          <div
            style="height: 0px;"
          >
            <div
              class="c4"
              data-strapi-header="true"
            >
              <div
                class="c5"
              >
                <a
                  class="c6"
                  href="/plugins/upload"
                  id="go-back"
                >
                  <svg
                    fill="none"
                    height="1rem"
                    viewBox="0 0 24 24"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M24 13.3a.2.2 0 0 1-.2.2H5.74l8.239 8.239a.2.2 0 0 1 0 .282L12.14 23.86a.2.2 0 0 1-.282 0L.14 12.14a.2.2 0 0 1 0-.282L11.86.14a.2.2 0 0 1 .282 0L13.98 1.98a.2.2 0 0 1 0 .282L5.74 10.5H23.8c.11 0 .2.09.2.2v2.6Z"
                      fill="#212134"
                    />
                  </svg>
                  <span
                    class="c7 c8"
                  >
                    Back
                  </span>
                </a>
              </div>
              <div
                class="c9"
              >
                <div
                  class="c10 c11"
                >
                  <h1
                    class="c7 c12"
                  >
                    Configure the view - Media Library
                  </h1>
                </div>
                <button
                  aria-disabled="true"
                  class="c13 c14 c15 c16"
                  disabled=""
                  type="submit"
                >
                  <div
                    aria-hidden="true"
                    class=""
                  >
                    <svg
                      fill="none"
                      height="1rem"
                      viewBox="0 0 24 24"
                      width="1rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20.727 2.97a.2.2 0 0 1 .286 0l2.85 2.89a.2.2 0 0 1 0 .28L9.554 20.854a.2.2 0 0 1-.285 0l-9.13-9.243a.2.2 0 0 1 0-.281l2.85-2.892a.2.2 0 0 1 .284 0l6.14 6.209L20.726 2.97Z"
                        fill="#212134"
                      />
                    </svg>
                  </div>
                  <span
                    class="c7 c17"
                  >
                    Save
                  </span>
                </button>
              </div>
              <p
                class="c7 c18"
              >
                Define the view settings of the media library.
              </p>
            </div>
          </div>
          <div
            class="c19"
          >
            <div
              class="c20"
            >
              <div
                class="c21"
              >
                <div
                  class="c22"
                >
                  <div
                    class=""
                  >
                    <div
                      class="c23"
                    >
                      <label
                        class="c7 c24 c25"
                        for=":r0:"
                      >
                        Entries per page
                      </label>
                      <div
                        aria-autocomplete="none"
                        aria-controls="radix-:r3:"
                        aria-describedby=":r0:-hint :r0:-error"
                        aria-expanded="false"
                        aria-label="Entries per page"
                        class="c26 c27 c28"
                        data-state="closed"
                        dir="ltr"
                        id=":r0:"
                        overflow="hidden"
                        role="combobox"
                        tabindex="0"
                      >
                        <span
                          class="c29 c30"
                        >
                          <span
                            class="c7 c31 c32"
                          >
                            <span
                              class="c33"
                            >
                              10
                            </span>
                          </span>
                        </span>
                        <span
                          class="c30"
                        >
                          <span
                            aria-hidden="true"
                            class="c34"
                          >
                            <svg
                              fill="none"
                              height="1rem"
                              viewBox="0 0 14 8"
                              width="1rem"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
                                fill="#32324D"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                      <p
                        class="c7 c35"
                        id=":r0:-hint"
                      >
                        Number of assets displayed by default in the Media Library
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="c22"
                >
                  <div
                    class=""
                  >
                    <div
                      class="c23"
                    >
                      <label
                        class="c7 c24 c25"
                        for=":r4:"
                      >
                        Default sort order
                      </label>
                      <div
                        aria-autocomplete="none"
                        aria-controls="radix-:r7:"
                        aria-describedby=":r4:-hint :r4:-error"
                        aria-expanded="false"
                        aria-label="Default sort order"
                        class="c26 c27 c28"
                        data-state="closed"
                        dir="ltr"
                        id=":r4:"
                        overflow="hidden"
                        role="combobox"
                        tabindex="0"
                      >
                        <span
                          class="c29 c30"
                        >
                          <span
                            class="c7 c31 c32"
                          >
                            <span
                              class="c33"
                            >
                              createdAt:DESC
                            </span>
                          </span>
                        </span>
                        <span
                          class="c30"
                        >
                          <span
                            aria-hidden="true"
                            class="c34"
                          >
                            <svg
                              fill="none"
                              height="1rem"
                              viewBox="0 0 14 8"
                              width="1rem"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
                                fill="#32324D"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                      <p
                        class="c7 c35"
                        id=":r4:-hint"
                      >
                        Note: You can override this value in the media library.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </main>
    </div>
  </div>
  <div
    class="c36"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
